### DISCUSSION TITLE
Field permissions for Users

### DISCUSSION THREAD

**Discussion by danieljwestman** [directus/directus#5019] [Archive]
Why does every user (without admin/update permissions) need to see all the fields in the User view?

In our case we only need avatar, name and email. I get that most of the fields is needed for App access, but why force to show them all for everybody?

![Skärmavbild 2021-04-13 kl  09 41 02](https://user-images.githubusercontent.com/20950568/114516060-5355c480-9c3d-11eb-810e-0e299d49107b.png)

**Comment by rijkvanzanten** [comment]
> I get that most of the fields is needed for App access, but why force to show them all for everybody?

That's the coupling here. In order to be able to read those fields, they need to be readable, and thus visible. I think it might be worth looking into having separate user permissions (maybe fixed based on app access) for the _current user_ vs "other" users 🤔 (cc @benhaynes).

**Reply by benhaynes** [reply]
Does our current permission structure allow for different permissions for YOU vs OTHERS? If so, I'm fine with this. Most of the current user's fields need to be readable for the App to work, but you only need a bare minimum for _other_ users. Here's my rough take on the minimum fields for other users:

* ID — I assume this is required since it's used on the junctions to users... but maybe not?
* First Name — if not provided, does the App fall back to something else?
* Last Name — if not provided, does the App fall back to something else?
* Avatar — similarly, we could have a fallback here in case you don't want avatars displayed
* Status — is this required by the App?
* Role — is this required by the App?

It seems like nothing should really be required for _other_ users, but we'll need to make sure all user references throughout the system have good fallbacks (hopefully all those references happen in consolidated parts of the codebase). Not sure what happens to the User Directory if you turn all these off though.

**Reply by rijkvanzanten** [reply]
> Does our current permission structure allow for different permissions for YOU vs OTHERS?

Not from a "regular" configuration perspective in the app, but we can do it through the app-minimal permissions.

As for the minimal fields you need from _other_ users, I guess technically speaking you don't need any. User references as m2o's will fall back to the minimal saved value, which would be the UUID of the user. Moving up from that, I would say (in order): email, name, avatar. Status/role of other users aren't used by the app.

**Reply by danieljwestman** [reply]
I like this discussion.

> That's the coupling here. In order to be able to read those fields, they need to be readable, and thus visible.

Would be great if possible to make fields hidden in directus_* collections (not necessary excluded from the system it self, only hidden for the user).

**Reply by paulleduc** [reply]
Indeed, this is a bit of a blocker, or at least a pain point in the app I'm developing with Directus Cloud. 

The use-case is the ability for users to have public profiles. A lot of the data in these profiles is populated by user fields such as first_name, last_name, avatar, etc. So all roles need view access only to these fields. But a user still needs viewing and updating permissions for all/most of the fields of their own user.

The hacky work-around I'm using to remedy this is to separate the public fields into a different "profile" collection with a one-to-one relationship to the user. All users can view all of its fields, while only the current user can update all of those fields. Additionally, only the current user can view and update their own user fields.

**Comment by docloulou** [comment]
Hi, any news about a built in features to separate /users with restricted fields and /users/me with all fields ?
